src/modules/library/document_models.py

```diff
@@ -0,0 +1,199 @@
+import numpy as np
+import operator
+import string
+from gensim.parsing.preprocessing import preprocess_string, strip_punctuation
+from nltk.corpus import stopwords as sw
+from sklearn.manifold import TSNE
+
+# embedding_2D (numpy.ndarray): a N by 2 matrix (N = number of documents) where i-th line
+#     is the reduced document embedding of i-th document in the list 'documents', prepared
+#     in order to plot the embedding.
+# embedding_3D (numpy.ndarray): a N by 3 matrix (N = number of documents) where i-th line
+#     is the reduced document embedding of i-th document in the list 'documents', prepared
+#     in order to plot the embedding.
+
+
+class DocumentModels:
+    """
+    A class to represent documents as document embeddings.
+
+    Attributes:
+        word_vectors (gensim.models.keyedvectors.*): word embedding we want to use to
+            construct document embeddings
+        documents (list(string)): list of documents we want to embed
+        stopwords (list(string)): list of words we want to remove from documents' texts.
+        embedding (numpy.ndarray): a matrix where i-th line is the document embedding of
+            i-th document in the list 'documents'.
+        embed_documents(): Creates the embedding for documents given at initialization and saves it in the attribute
+            'embedding'.
+
+    Methods:
+        tokenize(text): Tokenizes and removes stopwords from the provided text.
+        average_word_embedding(text): Creates a document embedding as the average of word embeddings of words that
+            appear in given text.
+        embed_list_of_documents(docs): Returns a document embedding of documents given as an argument.
+        add_documents(docs): Appends given documents to the model's attribute 'documents' and adds lines for document
+            embeddings of those documents at the end of the matrix 'embedding'.
+        remove_documents(docs): Removes given documents from the model's attribute 'documents' and removes lines for
+            their embeddings from the matrix 'embedding'.
+        reduce_dimension(dimension=2): Reduces the dimension of the embedding to 'dimension'.
+    """
+
+    def __init__(self, word_embedding, documents, stopwords=None):
+        """
+        Args:
+             word_embedding (gensim.models.keyedvectors.*): word embedding we want to use to
+                 construct document embeddings
+             documents (list(string)): list of documents we want to embed
+             stopwords (list(string)): list of words we want to remove from documents' texts.
+                 (Default = customized list of english stopwords from module nltk)
+        """
+
+        self.word_vectors = word_embedding
```

Comment:
Make the attributes private:
`self.word_vectors` -> `self.__word_vectors`
`self.documents` -> `self.__documents`
`self.stopwords` -> `self.__stopwords`
`self.embedding` -> `self.__embedding`

Rename the variables in the rest of the script.

Reply:
Done in all scripts.